lib/deimos/utils/db_poller/base.rb

```diff
@@ -128,10 +148,28 @@ def process_batch_with_span(batch, status)
           true
         end
 
+        # If multiple producers, loop through producers configured on poller subclass
+        # If single producer, @resource_class will be the constantized producer class
         # @param batch [Array<ActiveRecord::Base>]
         # @return [void]
         def process_batch(batch)
-          @producer.send_events(batch)
+          if @config.producer_class.present?
+            @resource_class.send_events(batch)
+          else
+            @resource_class.producers.each do |producer|
+              producer.send_events(batch)
+            end
+          end
+        end
+
+        # Configure log identifier and messages to be used in subclasses
+        # # @return [String]
+        def log_identifier
+          if @config.producer_class.present?
+            "#{self.class.name}: #{@resource_class.topic}"
+          else
+            "#{self.class.name}: #{@resource_class.producers.map(&:topic)}"
+          end
```

Comment:
I think there's something fundamental weird here... `@resource_class` should be `self` if we're using `poller_class` instead of `producer_class`. This is the base class, so we should just be able to reference `self.producers`.

Maybe we need an intermediate method, e.g.

```ruby
def producer_classes
  return self.producers self.producers.any?
  [@producer_class]
end
```

and then one more:
```ruby
def self.poller_class
  @producer_class || self
end
```

There should be very few `if` statements when we have those.

Reply:
@dorner Yes so the reason why I changed the code to use these if statements: `if @config.producer_class.present?` in order to determine if it should choose the producer or poller path, is because when I was testing using item-platform, I noticed that when I had `self.producers`, the self was the StateBased poller class, which would return an empty array since the producers method is actually setup on the inherited poller class. 
Not too sure if I am making sense or not.

Reply:
If that's the case, then we're missing a piece - [https://github.com/flipp-oss/deimos/blob/master/lib/deimos/utils/db_poller.rb#L15](https://github.com/flipp-oss/deimos/blob/master/lib/deimos/utils/db_poller.rb#L15) - we need to instantiate the correct class here rather than always use the parent class.

Reply:
@dorner This makes sense. I guess I can adjust the class_for_config method here: https://github.com/flipp-oss/deimos/blob/master/lib/deimos/utils/db_poller.rb#L26
and first check if poller_class is in the config and instantiate that class, and if not, then continue with the code already in that method.